Implement use of linting rule set for Blockchain Public Address API

**Problem description**
Implement use of linting rule set for Blockchain Public Address API.
what we have to do is well documented in Commonalities:
https://github.com/camaraproject/Commonalities/pull/110
https://github.com/camaraproject/Commonalities/pull/74

**Expected action**
Check our API with linting rule set
provide feedback to commonalities team


**Additional context**
cc: @rartych who asked project volunteer to perform this action.

LINTER RULES OUTPUT

> Run spectral lint code/API_definitions/blockchain_public_address.yaml --verbose --ruleset .spectral.yml
  spectral lint code/API_definitions/blockchain_public_address.yaml --verbose --ruleset .spectral.yml
  shell: /usr/bin/bash -e {0}
  env:
    APPLY_FIXES: all
    APPLY_FIXES_EVENT: pull_request
    APPLY_FIXES_MODE: commit
Found 65 rules (56 enabled)
Linting /home/runner/work/BlockchainPublicAddress/BlockchainPublicAddress/code/API_definitions/blockchain_public_address.yaml
warning  camara-reserved-words  Reserved words found in input: Consider avoiding the use of reserved word 'public'  /blockchain-public-addresses
warning  camara-reserved-words  Reserved words found in input: Consider avoiding the use of reserved word 'public'  /blockchain-public-addresses/{id}

/home/runner/work/BlockchainPublicAddress/BlockchainPublicAddress/code/API_definitions/blockchain_public_address.yaml
  46:7   warning  camara-path-param-id                Path Parameter Naming Warning: Use 'resource_id' instead of just 'id' in path parameters.   paths
  47:3[2](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7645000818/job/20830571543?pr=1#step:6:2)    error  camara-parameter-casing-convention  /blockchain-public-addresses should be kebab-case: must be kebab case                       paths./blockchain-public-addresses
  58:1[3](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7645000818/job/20830571543?pr=1#step:6:3)  warning  oas3-operation-security-defined     "blockchain-public-address:read" must be listed among scopes.                               paths./blockchain-public-addresses.get.security[0].openId[0]
 11[4](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7645000818/job/20830571543?pr=1#step:6:4):13  warning  oas3-operation-security-defined     "blockchain-public-address:create" must be listed among scopes.                             paths./blockchain-public-addresses.post.security[0].openId[0]
 1[5](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7645000818/job/20830571543?pr=1#step:6:5)2:37    error  camara-parameter-casing-convention  /blockchain-public-addresses/{id} should be kebab-case: must be kebab case                  paths./blockchain-public-addresses/{id}
 1[6](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7645000818/job/20830571543?pr=1#step:6:6)3:13  warning  oas3-operation-security-defined     "blockchain-public-address:delete" must be listed among scopes.                             paths./blockchain-public-addresses/{id}.delete.security[0].openId[0]
 191:12  warning  camara-parameters-descriptions      Parameter description is missing or empty: "openId.description" property must be truthy     components.securitySchemes.openId
 20[7](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7645000818/job/20830571543?pr=1#step:6:7):13  warning  camara-path-param-id                Path Parameter Naming Warning: Use 'resource_id' instead of just 'id' in path parameters.   components.parameters.Id.name
 311:15  warning  camara-parameters-descriptions      Parameter description is missing or empty: "ErrorInfo.description" property must be truthy  components.schemas.ErrorInfo
 344:21    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.InvalidArgumentForBlockchain400.content.application/json.examples.InvalidArgument.value
 34[8](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7645000818/job/20830571543?pr=1#step:6:9):[21](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7645000818/job/20830571543?pr=1#step:6:22)    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.InvalidArgumentForBlockchain400.content.application/json.examples.InvalidBlockchainNetworkId.value
 352:21    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.InvalidArgumentForBlockchain400.content.application/json.examples.InvalidCurrencyForBlockchain.value
 356:21    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.InvalidArgumentForBlockchain400.content.application/json.examples.RequiredCurrencyforBlockchain.value
 373:21    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.PermissionDeniedForBlockchain403.content.application/json.examples.PermissionDenied.value
 377:21    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.PermissionDeniedForBlockchain403.content.application/json.examples.NotAllowedBlockchainNetworkId.value

After Linter execution with configuration as per: https://github.com/camaraproject/Commonalities/pull/110/commits/29e4e9da8b8ca7dd0e435fb8a942ac4ba3952b59

OutPut:

`Run spectral lint code/API_definitions/blockchain_public_address.yaml --verbose --ruleset .spectral.yml
Found 65 rules (56 enabled)
Linting /home/runner/work/BlockchainPublicAddress/BlockchainPublicAddress/code/API_definitions/blockchain_public_address.yaml

/home/runner/work/BlockchainPublicAddress/BlockchainPublicAddress/code/API_definitions/blockchain_public_address.yaml
  46:7   warning  camara-path-param-id                Path Parameter Naming Warning: Use 'resource_id' instead of just 'id' in path parameters.   paths
  47:32    error  camara-parameter-casing-convention  /blockchain-public-addresses should be kebab-case: must be kebab case                       paths./blockchain-public-addresses
  5[8](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7759340859/job/21163172764?pr=1#step:6:9):13  warning  oas3-operation-security-defined     "blockchain-public-address:read" must be listed among scopes.                               paths./blockchain-public-addresses.get.security[0].openId[0]
 114:13  warning  oas3-operation-security-defined     "blockchain-public-address:create" must be listed among scopes.                             paths./blockchain-public-addresses.post.security[0].openId[0]
 152:37    error  camara-parameter-casing-convention  /blockchain-public-addresses/{id} should be kebab-case: must be kebab case                  paths./blockchain-public-addresses/{id}
 163:13  warning  oas3-operation-security-defined     "blockchain-public-address:delete" must be listed among scopes.                             paths./blockchain-public-addresses/{id}.delete.security[0].openId[0]
 1[9](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7759340859/job/21163172764?pr=1#step:6:10)1:12  warning  camara-parameters-descriptions      Parameter description is missing or empty: "openId.description" property must be truthy     components.securitySchemes.openId
 207:13  warning  camara-path-param-id                Path Parameter Naming Warning: Use 'resource_id' instead of just 'id' in path parameters.   components.parameters.Id.name
 3[11](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7759340859/job/21163172764?pr=1#step:6:12):[15](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7759340859/job/21163172764?pr=1#step:6:16)  warning  camara-parameters-descriptions      Parameter description is missing or empty: "ErrorInfo.description" property must be truthy  components.schemas.ErrorInfo
 344:[21](https://github.com/PedroDiez/BlockchainPublicAddress/actions/runs/7759340859/job/21163172764?pr=1#step:6:22)    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.InvalidArgumentForBlockchain400.content.application/json.examples.InvalidArgument.value
 348:21    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.InvalidArgumentForBlockchain400.content.application/json.examples.InvalidBlockchainNetworkId.value
 352:21    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.InvalidArgumentForBlockchain400.content.application/json.examples.InvalidCurrencyForBlockchain.value
 356:21    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.InvalidArgumentForBlockchain400.content.application/json.examples.RequiredCurrencyforBlockchain.value
 373:21    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.PermissionDeniedForBlockchain403.content.application/json.examples.PermissionDenied.value
 377:21    error  oas3-valid-media-example            "value" property must have required property "status"                                       components.responses.PermissionDeniedForBlockchain403.content.application/json.examples.NotAllowedBlockchainNetworkId.value`

--- a/code/API_definitions/blockchain_public_address.yaml
+++ b/code/API_definitions/blockchain_public_address.yaml
@@ -342,18 +342,22 @@ components:
           examples:
             InvalidArgument:
               value:
+                status: 400
                 code: INVALID_ARGUMENT
                 message: Client specified an invalid argument, request body or query param
             InvalidBlockchainNetworkId:
               value:
+                status: 400
                 code: BLOCKCHAIN_PUBLIC_ADDRESS.INVALID_BLOCKCHAIN_NETWORK_IDENTIFIER
                 message: Indicated blockchainNetworkId is not valid
             InvalidCurrencyForBlockchain:
               value:
+                status: 400
                 code: BLOCKCHAIN_PUBLIC_ADDRESS.INVALID_CURRENCY
                 message: Indicated currency is not found
             RequiredCurrencyforBlockchain:
               value:
+                status: 400
                 code: BLOCKCHAIN_PUBLIC_ADDRESS.CURRENCY_REQUIRED
                 message: One currency is required
     PermissionDeniedForBlockchain403:
@@ -371,10 +375,12 @@ components:
           examples:
             PermissionDenied:
               value:
+                status: 403
                 code: PERMISSION_DENIED
                 message: "Operation not allowed: ..."
             NotAllowedBlockchainNetworkId:
               value:
+                status: 403
                 code: BLOCKCHAIN_PUBLIC_ADDRESS.BLOCKCHAIN_NETWORK_IDENTIFIER_NOT_ALLOWED
                 message: Indicated blockchainNetworkId is not allowed by business rules
     Generic400:

